lib/compress/zstd_compress.c

```diff
@@ -5645,8 +5669,27 @@ size_t ZSTD_compressStream2( ZSTD_CCtx* cctx,
 
     /* transparent initialization stage */
     if (cctx->streamStage == zcss_init) {
-        FORWARD_IF_ERROR(ZSTD_CCtx_init_compressStream2(cctx, endOp, input->size), "CompressStream2 initialization failed");
```

Comment:
As you point out in the comment `inputSize = input->size - input->pos`. So this is actually a bug, and we'll get the `pledgedSrcSize` wrong.

Can you add a test case for the scenario where `input->pos` starts off as non-zero? It should fail before the PR.

Maybe you've already added that test case, but I haven't gotten to the tests yet.

Reply:
We won't get `pledgedSrcSize` wrong because it's based on `totalInputSize`, which correctly counts from initial `pos`, wether it is 0 or non-0, by summing the `inputSize` across invocations.

Point taken for the `input->pos != 0` start condition test.

Reply:
I'm talking about the previous version of the code. It is buggy before your patch, and correct after.